File: packages/shared-types/src/listing.ts
export interface Platform {
  id: number;
  name: string;           // 'liveauctioneers', 'estatesales_net', etc.
  display_name: string;   // 'LiveAuctioneers', 'EstateSales.NET', etc.
  base_url: string;
  logo_url: string | null;
}

export type ListingCondition = "excellent" | "good" | "fair" | "poor";

export interface Listing {
  id: number;
  platform: Platform;
  external_id: string;
  /** Always redirect here - we never handle transactions directly */
  external_url: string;
  title: string;
  description: string | null;
  category: string | null;
  condition: ListingCondition | null;

  // Pricing
  current_price: number | null;
  final_price: number | null;
  is_completed: boolean;
  /** e.g. 25 means 25% buyer's premium on top of the hammer price */
  buyers_premium_pct: number | null;
  /** current_price * (1 + buyers_premium_pct/100) - shown as "your total cost" */
  total_cost_estimate: number | null;

  // Fulfillment
  pickup_only: boolean;
  ships_nationally: boolean;

  // Location
  city: string | null;
  state: string | null;
  zip_code: string | null;
  latitude: number | null;
  longitude: number | null;

  // Timing (ISO 8601 strings)
  sale_ends_at: string | null;
  sale_starts_at: string | null;

  // Media
  primary_image_url: string | null;
  image_urls: string[];

  scraped_at: string;

  // Computed by API
  distance_miles?: number;
  is_sponsored?: boolean;
}

export interface SearchFilters {
  q?: string;
  lat?: number;
  lon?: number;
  /** Default 50, max 500 */
  radius_miles?: number;
  min_price?: number;
  max_price?: number;
  pickup_only?: boolean;
  /** Items ending within N hours */
  ending_hours?: number;
  category?: string;
  platform_ids?: number[];
  page?: number;
  page_size?: number;
}

export const CATEGORIES = [
  "ceramics",
  "furniture",
  "jewelry",
  "art",
  "silver",
  "glass",
  "books",
  "clothing",
  "tools",
  "electronics",
  "collectibles",
  "other",
] as const;

export type Category = (typeof CATEGORIES)[number];

File: backend/app/schemas/listing.py
from datetime import datetime
from pydantic import BaseModel, Field


class PlatformOut(BaseModel):
    id: int
    name: str
    display_name: str
    base_url: str
    logo_url: str | None

    model_config = {"from_attributes": True}


class ListingOut(BaseModel):
    id: int
    platform: PlatformOut
    external_id: str
    external_url: str
    title: str
    description: str | None
    category: str | None
    condition: str | None
    listing_type: str = "auction"
    item_type: str = "individual_item"
    auction_status: str = "upcoming"
    current_price: float | None
    final_price: float | None
    is_completed: bool
    buyers_premium_pct: float | None
    total_cost_estimate: float | None
    pickup_only: bool
    ships_nationally: bool
    city: str | None
    state: str | None
    zip_code: str | None
    latitude: float | None
    longitude: float | None
    sale_ends_at: datetime | None
    sale_starts_at: datetime | None
    primary_image_url: str | None
    image_urls: list[str]
    scraped_at: datetime
    # Enriched structured fields (populated by enricher.py at ingest time)
    maker: str | None = None
    brand: str | None = None
    period: str | None = None
    country_of_origin: str | None = None
    sub_category: str | None = None
    attributes: dict = Field(default_factory=dict)
    # computed
    distance_miles: float | None = None
    is_sponsored: bool = False

    model_config = {"from_attributes": True}


class ListingLotOut(BaseModel):
    id: int
    listing_id: int
    lot_number: str | None
    title: str
    description: str | None
    category: str | None
    condition: str | None
    current_price: float | None
    hammer_price: float | None
    estimate_low: float | None
    estimate_high: float | None
    is_completed: bool
    bid_count: int | None
    sale_ends_at: datetime | None
    primary_image_url: str | None
    image_urls: list[str]
    external_url: str | None
    scraped_at: datetime

    model_config = {"from_attributes": True}


class ListingSearchParams(BaseModel):
    q: str | None = None
    lat: float | None = None
    lon: float | None = None
    radius_miles: int = Field(50, ge=1, le=500)
    min_price: float | None = Field(None, ge=0)
    max_price: float | None = None
    pickup_only: bool = False
    ending_hours: int | None = None
    category: str | None = None
    platform_ids: list[int] = Field(default_factory=list)
    listing_type: str | None = None
    page: int = Field(1, ge=1)
    page_size: int = Field(24, ge=1, le=100)
